### PR TITLE
[ADD] search method for partner age

### PR DESCRIPTION
**Pull request by victor-champonnois**
This PR adds a `_search_age` method for partner's `age` computed field. Tests are also added.

### CODE REVIEW

**Review by robinkeunen — APPROVED** [review]

File: partner_contact_birthdate/models/res_partner.py [hunk 1]
```diff
@@ -23,3 +25,20 @@ def _compute_age(self):
...
+        # the value of operator is checked, no risk of injection
+        query = """
+            SELECT id
+            FROM res_partner
+            WHERE extract(year from age(CURRENT_DATE, birthdate_date))
+              {operator} %s
+            """.format(
+            operator=operator
+        )
+        self.env.cr.execute(query, (value,))
```
Comment (carmenbianca) [inline partner_contact_birthdate/models/res_partner.py]: I don't think `in` and `not in` are valid SQL. They should be removed from the list of operators.

**Review by carmenbianca — CHANGES_REQUESTED** [review]

File: partner_contact_birthdate/models/res_partner.py (hunk 1 above)
Comment (victor-champonnois) [reply partner_contact_birthdate/models/res_partner.py]: It exists in [PostgreSQL's doc ](https://www.postgresql.org/docs/current/functions-subquery.html#FUNCTIONS-SUBQUERY-IN), why do you think it's not valid?

Comment (carmenbianca) [reply partner_contact_birthdate/models/res_partner.py]: Huh, you may be right. Can you maybe add a test for them?

Comment (carmenbianca) [reply partner_contact_birthdate/models/res_partner.py]: On second inspection: I think they don't work. The operators require a subquery.

Comment (victor-champonnois) [reply partner_contact_birthdate/models/res_partner.py]: OK let's just remove them.

**Review by carmenbianca — APPROVED** [review]
LGTM! :tada:

**Pull request merged**